torch/csrc/api/include/torch/nn/functional/pooling.h

```diff
@@ -776,8 +776,8 @@ inline std::tuple<Tensor, Tensor> fractional_max_pool2d_with_indices(
 
   Tensor _random_samples_ = _random_samples;
   if (!_random_samples_.defined()) {
-    auto n_batch = input.dim() == 3;
```

Comment:
This is incorrect because it will always set to either 0 or 1. The implementation in [the diff prior to this change](https://github.com/pytorch/pytorch/pull/62490/files) was 

`auto n_batch = 1 ? input.dim() == 3 : input.size(0)`

which is obviously a bug, so I've changed it to the correct code and added an appropriate test above to test for not just the no_batch_dim case.

Reply:
nice work!